### ISSUE TITLE
Recreating mappings from original index does not work

### ISSUE THREAD

**Issue by Enzo90910** [pgaertig/es-reindex#2]
When running against my ES 1.2.1 with -r option, I get
Retrying PUT ERROR: RestClient::BadRequest - 400 Bad Request
when the script tries to recreate the mappings in the destination index.
I traced the problem to the line
mappings[sidx].each_pair{|type, mapping|
On my machine, mappings[sidx] does not directly contain a dictionary of mappings, but a "mappings" object which itself contains the dictionary of mappings.
I solved it by replacing said line with 
mappings[sidx]["mappings"].each_pair{|type, mapping|
which seems to have solved the problem.

**Comment by kubagruszka** [comment]
I've encoutered the same issue while trying to migrate index. Applying @JustinAiken fix helped me.

**Comment by Cylindric** [comment]
Yep, I had the same problem and that fix worked for me too. Thanks @JustinAiken!

**Comment by JustinAiken** [comment]
Glad it's working for you guys! :smile:
